lib/portage/package/ebuild/doebuild.py

```diff
@@ -2877,6 +2911,11 @@ def _post_src_install_soname_symlinks(mysettings, out):
         if f is not None:
             f.close()
 
+    # We do RDEPEND mangling here instead of the natural location
+    # in _post_src_install_write_metadata because NEEDED hasn't been
+    # written yet at that point.
+    _inject_libc_dep(build_info_dir, mysettings)
```

Comment:
I feel like this should at least be conditional on the presence of an ELF file. Even better if we check for an soname dependency that is satisfied by libc.

Reply:
It should be already here -- we only get to that point if there's a `NEEDED` -> we have an ELF file. The function returns early above it otherwise.

(But the previous behaviour of this PR was to not do that - you are right it's important).

Let me know if I'm missing something here though

Reply:
Okay, looks good. Thanks!